### PR TITLE
Bobax 35 modal quantity milk option

### PR DESCRIPTION
**Pull request by SimonMTan**
Create title/description/Milk options with Radio Button using React-bootstrap while maintain the code to be reusable/scale able

### CODE REVIEW

File: src/assets/sampleBobaAPI.tsx [hunk 1]
```diff
@@ -155,23 +164,23 @@ export const bobaList: BobaProps[] = [
...
-  { name: 'Aloe Vera', price: 0.75 },
-  { name: 'Lychee Jelly', price: 0.75 },
-  { name: 'Coconut Jelly', price: 0.75 },
-  { name: 'Red Bean', price: 0.75 },
-  { name: 'Mango Jelly', price: 0.75 },
-  { name: 'Pudding', price: 0.5 },
-  { name: 'Milk Foam', price: 1.0 },
-  { name: 'Jelly', price: 0.5 },
-];
+export const toppingsList: ToppingList = {
```
Comment (seintun) [inline src/assets/sampleBobaAPI.tsx]: Why is the name changed? `ToppingsProps`

File: src/assets/sampleBobaAPI.tsx [hunk 2]
```diff
@@ -155,23 +164,23 @@ export const bobaList: BobaProps[] = [
...
-  { name: 'Almond Milk' },
-  { name: 'Soy Milk' },
-  { name: 'Oat Milk' },
-];
+export const milkList: MilkList = {
+  milk1: { name: 'Whole Milk', price: 0.0 },
+  milk2: { name: '2% Milk', price: 0.5 },
+  milk3: { name: 'Almond Milk', price: 0.5 },
+  milk4: { name: 'Soy Milk', price: 0.5 },
+  milk5: { name: 'Oat Milk', price: 0.5 },
```
Comment (seintun) [inline src/assets/sampleBobaAPI.tsx]: `price` was intentionally not added to reduce complexity

File: src/components/Header/Heading.tsx
Comment (seintun) [inline src/components/Header/Heading.tsx:1]: I think you added this back from merge conflict resolution.
`Heading` was removed and not used anywhere

File: src/assets/sampleBobaAPI.tsx [hunk 3]
```diff
@@ -18,6 +18,15 @@ interface ToppingsProps {
...
 interface MilkProps {
   name: string;
+  price: number;
+}
+
+interface MilkList {
+  [key: string]: MilkProps;
+}
+
+interface ToppingList {
```
Comment (seintun) [inline src/assets/sampleBobaAPI.tsx]: `ToppingListProps`

File: src/assets/sampleBobaAPI.tsx [hunk 4]
```diff
@@ -18,6 +18,15 @@ interface ToppingsProps {
 
 interface MilkProps {
   name: string;
+  price: number;
+}
+
+interface MilkList {
```
Comment (seintun) [inline src/assets/sampleBobaAPI.tsx]: `MilkListProps`

File: src/types/common/main.d.ts [hunk 5]
```diff
@@ -20,3 +20,27 @@ export interface CustomizeBobaModalProps {
...
+}
+
+export interface MilkList {
+  [key: string]: {
+    name: string;
+    price: number;
+  };
+}
+
+export interface ToppingList {
```
Comment (seintun) [inline src/types/common/main.d.ts]: `ToppingListProps`
If these exist and available, why not import into `src/assets/sampleBobaAPI.tsx` for use?

File: src/components/Modal/index.tsx [hunk 6]
```diff
@@ -38,11 +38,11 @@ const BaseModal: React.FC<BaseModalProps> = ({ title, modalShow, onHide, ModalBo
...
-  modalShow: PropTypes.bool.isRequired,
-  onHide: PropTypes.func.isRequired,
-  ModalBody: PropTypes.func,
-};
+// BaseModal.propTypes = {
+//   title: PropTypes.string.isRequired,
+//   modalShow: PropTypes.bool.isRequired,
+//   onHide: PropTypes.func.isRequired,
+//   ModalBody: PropTypes.func,
+// };
```
Comment (seintun) [inline src/components/Modal/index.tsx]: Delete if commented

File: src/pages/Home/index.tsx [hunk 7]
```diff
@@ -5,6 +5,7 @@ import { bobaList } from 'assets/sampleBobaAPI';
 import BobaContainer from 'components/BobaContainer';
 import SearchBar from 'components/SearchBar';
 import CustomizeModal from 'components/Modal';
+import BobaModalform from 'pages/Home/CustomizeBobaModalBody';
```
Comment (seintun) [inline src/pages/Home/index.tsx]: use `./CustomizeBobaModalBody` if they're in the same directory

File: src/types/common/main.d.ts (hunk 5 above)
Comment (SimonMTan) [reply src/types/common/main.d.ts]: i just posted there since sampleBobaAPI will go away eventually

File: src/assets/sampleBobaAPI.tsx [hunk 8]
```diff
@@ -13,11 +13,20 @@ export interface BobaProps {
 
 interface ToppingsProps {
   name: string;
-  price: number;
+  // price: number;
```
Comment (seintun) [inline src/assets/sampleBobaAPI.tsx]: Remove any unused commented code before merging into develop

File: src/assets/sampleBobaAPI.tsx [hunk 9]
```diff
@@ -13,11 +13,20 @@ export interface BobaProps {
...
 }
 
 interface MilkProps {
   name: string;
+  // price: number;
```
Comment (seintun) [inline src/assets/sampleBobaAPI.tsx]: Remove any unused commented code before merging into develop

File: src/types/common/main.d.ts [hunk 10]
```diff
@@ -20,3 +20,27 @@ export interface CustomizeBobaModalProps {
...
+
+export interface MilkProps {
+  name: string;
+  price: number;
+}
+
+export interface MilkListProps {
+  [key: string]: {
+    name: string;
+    // price: number;
```
Comment (seintun) [inline src/types/common/main.d.ts]: Remove any unused commented code before merging into develop

File: src/types/common/main.d.ts [hunk 11]
```diff
@@ -20,3 +20,27 @@ export interface CustomizeBobaModalProps {
...
 }
+
+export interface ToppingsProps {
+  name: string;
+  price: number;
+}
+
+export interface MilkProps {
+  name: string;
+  price: number;
```
Comment (seintun) [inline src/types/common/main.d.ts]: Remove price

File: src/types/common/main.d.ts (hunk 5 above)
Comment (seintun) [reply src/types/common/main.d.ts]: Okay please use ToppingListProps

File: src/types/common/main.d.ts [hunk 12]
```diff
@@ -20,3 +20,27 @@ export interface CustomizeBobaModalProps {
   modalShow: boolean;
   onHide: () => void;
 }
+
+export interface ToppingsProps {
```
Comment (seintun) [inline src/types/common/main.d.ts]: Nit: Remove s from Toppings

File: src/pages/Home/CustomizeBobaModalBody.tsx [hunk 13]
```diff
@@ -0,0 +1,32 @@
+import Form from 'react-bootstrap/Form';
+import Row from 'react-bootstrap/Row';
+import Col from 'react-bootstrap/Col';
+import { Container } from 'react-bootstrap';
+import React from 'react';
```
Comment (seintun) [inline src/pages/Home/CustomizeBobaModalBody.tsx]: Import react first

File: src/components/RadioButton/index.tsx [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+              checked={milk === milkOption.name}
+            />
+          </div>
+        );
+      })}
+    </>
+  );
+};
+
+export default CustomizeRadioButton;
```
Comment (seintun) [inline src/components/RadioButton/index.tsx:35]: RadioButton is fine

File: src/components/Modal/index.tsx [hunk 15]
```diff
@@ -1,6 +1,6 @@
 import Button from 'react-bootstrap/Button';
 import Modal from 'react-bootstrap/Modal';
-import PropTypes from 'prop-types';
+// import PropTypes from 'prop-types';
```
Comment (seintun) [inline src/components/Modal/index.tsx]: Remove if unused and commented

File: src/assets/sampleBobaAPI.tsx [hunk 16]
```diff
@@ -13,11 +13,20 @@ export interface BobaProps {
 
 interface ToppingsProps {
```
Comment (seintun) [inline src/assets/sampleBobaAPI.tsx]: Not: Remove s from Toppings

**Review by seintun — APPROVED** [review]

**Pull request merged**